src/java/main/org/apache/zookeeper/server/quorum/QuorumCnxManager.java

```diff
@@ -660,6 +665,9 @@ public void run() {
                             "Ignoring exception", ie);
                     }
                     closeSocket(client);
+                } catch (ConfigException ce) {
+                    LOG.error(ce.getMessage());
+                    throw new RuntimeException(ce);
```

Comment:
You are right. Throwing 'RuntimeException' does skip the closing `ServerSocket` part, so I moved it to the end of the` run` method.